src/adapter/src/coord/message_handler.rs

```diff
@@ -99,7 +99,7 @@ impl<S: Append + 'static> Coordinator<S> {
         let internal_cmd_tx = self.internal_cmd_tx.clone();
         let client = self.storage_usage_client.clone();
         // Similar to audit events, use the oracle ts so this is guaranteed to increase.
-        let collection_timestamp: EpochMillis = self.get_local_write_ts().await.timestamp.into();
+        let collection_timestamp: EpochMillis = self.peek_local_write_ts().into();
```

Comment:
I'm not entirely convinced that this is the right approach. `peek_local_write_ts` is good for looking at what the current timestamp is, but not really good for getting a timestamp that will get written down somewhere. The value returned by `peek_local_write_ts` is not guaranteed to be unique and therefore you may have some other event (audit event or write to Storage) with the exact same timestamp. 

`get_local_write_ts` is guaranteed to always give a unique timestamp and is generally what we use when that timestamp will be written down. I'm not 100% convinced that it's important for this timestamp to be unique, but I think it's probably on the safer side to use a unique timestamp.